developers-map.md

```diff
@@ -0,0 +1,171 @@
+# Developer Map
+
+This document is meant to provide a mapping from the conceptual model of Secretless to the codebase.
+
+## Directories
+
++ [./cmd](./cmd) - entrypoints for [binaries](#binaries)
++ [./internal/app](./internal/app) - internal implementations of [plugins](#plugins)
++ [./internal/pkg](./internal/pkg) - [plugin manager](#pluginmanager) and [resolver](#resolver)
++ [./pkg/secretless](./pkg/secretless) - secretless [config](#config) and [plugin interfaces](#plugins)
+
+## Binaries
+
+The entrypoints to binaries exist in "./cmd/${binary_name}"
+
+2 binaries are provided in the secretless repo.
++ [secretless](#secretless)
++ summon2
+
+## Secretless
+
+### Configuration
+config.Config
+[./pkg/secretless/config/config.go](./pkg/secretless/config/config.go)
+
++ represents the secretless configuration as struct
+
+### Entrypoint
+[./cmd/secretless-broker/main.go](./cmd/secretless-broker/main.go)
+
+At the entrpoiny, some parameters are obtained via flags:
+
++ -f secretless config file
++ -p plugin directory
++ --watch condition to reload when config file changes
++ --debug toggles debug information
++ --config-mgr strategy to provide secretless configuration
+
+Execution shifts to the [Plugin Manager](#pluginmanager):
+  + load internal [plugins](#plugins)
+  + load external [plugins](#plugins) from the plugin directory
+  + registers signal handlers
+  + run
+
+### PluginManager 
+plugin.Manager
+
+[./internal/pkg/plugin/manager.go](./internal/pkg/plugin/manager.go)
+
++ is a singleton
++ loads and holds [plugins](#plugins)
++ creates
+  + [Resolver](#resolver)
+  + [Proxy](#proxy)
++ root of
+  + RunHandlerFunc
+  + RunListenerFunc
++ manages all the [plugins](#plugins) 
++ initialises and runs [Proxy](#proxy) 
++ implements [EventNotifier](#eventnotifier) (only one in the whole repo)
+
+
+## Plugins
+Plugins are used as a mechanism for extending Secretless functionality beyond the core. These are expressed as interfaces. They are versioned in anticipation of growth.
```

Comment:
To be more exact, plugins are used also to provide core functionality - without internal plugins, secretless by itself doesn't work.